helper/terraformtype/helper/schema/attributes.go

```diff
@@ -59,3 +60,42 @@ func ParseAttributeReference(reference string) ([]string, error) {
 
 	return attributeReferenceParts, nil
 }
+
+// ValidateAttributeReference validates schema attribute reference.
+// Attribute references are used in Schema fields such as AtLeastOneOf, ConflictsWith, and ExactlyOneOf.
+func ValidateAttributeReference(tfresource *tfschema.Resource, reference string) ([]string, error) {
+	var curSchemaOrResource interface{} = tfresource
+	attributeReferenceParts := strings.Split(reference, ".")
+	for idx, attributeReferencePart := range attributeReferenceParts {
+		attributeReference := strings.Join(attributeReferenceParts[:idx+1], ".")
+		if math.Mod(float64(idx), 2) == 1 {
+			// For odd part, ensure it is a 0 and the containing schema has `MaxItems` set to `1`. This is because
+			// reference among multiple instances of the same nested block is not supported in current plugin SDK.
+			attributeReferencePartInt, err := strconv.Atoi(attributeReferencePart)
+
+			configurationBlockReferenceErr := fmt.Errorf("%q configuration block attribute references must be separated by .0", attributeReference)
+			if err != nil {
+				return nil, configurationBlockReferenceErr
+			}
+			if attributeReferencePartInt != 0 {
+				return nil, configurationBlockReferenceErr
+			}
+
+			curSchema := curSchemaOrResource.(*tfschema.Schema)
+			if curSchema.MaxItems != 1 || curSchema.Type != tfschema.TypeList {
+				return nil, fmt.Errorf("%q configuration block attribute references are only valid for TypeList and MaxItems: 1 attributes", attributeReference)
+			}
+			curSchemaOrResource = curSchema.Elem
+		} else {
+			// For even part, ensure it references to defined attribute
+			schema := curSchemaOrResource.(*tfschema.Resource).Schema[attributeReferencePart]
+			if schema == nil {
+				return nil, fmt.Errorf("%q references to unknown attribute", attributeReference)
+			}
+			curSchemaOrResource = schema
```

Comment:
Same here. We will want to be careful for unexpected types and missing key references:

```suggestion
			// For even part, ensure it references to defined attribute
			switch current := curSchemaOrResource.(type) {
			case *tfschema.Resource:
				if current.Schema == nil {
					return nil, fmt.Errorf("%q references attribute without schema")
				}

				schema, ok := current.Schema[attributeReferencePart]

				if !ok || schema == nil {
					return nil, fmt.Errorf("%q references to unknown attribute", attributeReference)
				}

				curSchemaOrResource = schema
			}
```

Reply:
A little improvement of this might be just using `schema := current.Schema[attributeReferencePart]` instead, and in absent key case, it will return the zero value of the Value, in which case it is `nil` (since we do not differ between absent key and nil value).